### PR TITLE
Convert sphe to C++

### PR DESCRIPTION
**Pull request by maxpkatz**
## PR checklist

- [x] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Comment by zingale** [comment]
radiation tests pass:
http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2023-09-24/index.html

File: Source/radiation/RadSolve.cpp [hunk 1]
```diff
@@ -1160,8 +1169,17 @@ void RadSolve::getEdgeMetric(int idim, const Geometry& geom, const Box& edgebox,
         geom.GetEdgeLoc(s, reg, I);
       }
       const Real *dx = geom.CellSize();
-      sphe(r.dataPtr(), s.dataPtr(), idim,
-           ARLIM(edgebox.loVect()), ARLIM(edgebox.hiVect()), dx);
+
```
Comment (zingale) [inline Source/radiation/RadSolve.cpp:1172]: aren't we missing the 1-d case here, r = r**2?

Comment (zingale) [reply Source/radiation/RadSolve.cpp:1172]: and the n = 0 case for multid?

Comment (maxpkatz) [reply Source/radiation/RadSolve.cpp:1172]: fixed, I think

**Review by zingale — APPROVED** [review]

**Pull request merged**